react-backend/client/src/components/admin/adminVisitsPage.js

```diff
@@ -46,9 +82,31 @@ export class AdminVisitsPage extends Component {
       });
   };
 
-  filter = e => {
-    const filterBy = e.target.value;
-    this.updateResults(filterBy);
+  sort = e => {
+    this.setState(
+      {
+        orderBy: e.target.value
+      },
+      this.updateResults
+    );
+  };
+
+  filter = (group, e) => {
```

Comment:
Rather than binding as below, I would prefer this to be partially applied like in my example I descibed before:

```js
filter = group => e => {//..the rest
```

That would mean you don't need to bind below, you just call the function with the group:

```js
 onChange={this.filter('gender')}
```
vs
```js
 onChange={this.filter.bind(this, 'gender')}
```